torch/autograd/_functions/reduce.py

```diff
@@ -71,18 +72,19 @@ def backward(self, grad_output):
             grad_input[some_zeros] = 0
 
             single_zero_idx = slice_zero_count.eq(1).nonzero()
-            for idx in single_zero_idx:
-                idx_tuple = tuple(idx.cpu())
-                input_idx_tuple = idx_tuple[:self.dim] + (slice(0, None),) + idx_tuple[self.dim + 1:]
-
-                # slice_mask and input_copy are 1D
-                slice_mask = zero_mask[input_idx_tuple]
-                input_copy = input[input_idx_tuple].clone()
-                zero_idx = slice_mask.nonzero()[0, 0]
-                input_copy[zero_idx] = 1.
-
-                grad_idx_tuple = idx_tuple[:self.dim] + (zero_idx,) + idx_tuple[self.dim + 1:]
-                grad_input[grad_idx_tuple] = grad_output[idx_tuple] * input_copy.prod()
+            if len(single_zero_idx) > 0:
```

Comment:
In theory yes but if `single_zero_idx` is an empty tensor, trying to iterate through it raises a `RuntimeError` because "dimension 0 out of range of 0D tensor".